src/lumigo_tracer/wrappers/http/sync_http_wrappers.py

```diff
@@ -224,7 +231,7 @@ def _requests_wrapper(func, instance, args, kwargs):
             method = safe_get_list(args, 0, kwargs.get("method", "")).upper()
             url = safe_get_list(args, 1, kwargs.get("url"))
             if HttpState.previous_request and HttpState.previous_request.host in url:
-                span = SpansContainer.get_span().get_last_span()
+                span = SpansContainer.get_span().get_span_by_id(HttpState.previous_span_id)
```

Comment:
It's not the same as it was before? taking the last span that was sent and not the current one?

Reply:
This is the only case that we're using this "last one" - this is relevant only to an extremely specific case wherein the `requests` package failed in some outer-processing logic (i.e. not part of the HTTP request).
In this case, I don't have any other id that can redirect me to the request itself (instance is None), so I'm using a heuristic of validating the URL 🤷

Reply:
I think in this case the best is to send "half" span, this is just a mistake and the user can't notice that

Reply:
Don't you have `instance` so you should have the id you attached to the `HTTPConnection` object?

Reply:
ok, we will do this `HttpState.previous_span_id` only if we have a flag of IS_TRACER_SYNC 👌